commit 9bb61afb12a14ac5ff3a8fc9a9def9487f371c68
Author: j_r0dd <285797+jr0dd@users.noreply.github.com>
Date:   Mon Sep 13 10:15:36 2021 -0400

    [prometheus-nut-exporter] metrics refactor (#1177)
    
    * [prometheus-nut-exporter] metrics refactor
    
    * fix version
    
    * add template to rules
    
    * Update charts/stable/prometheus-nut-exporter/templates/prometheusrules.yaml
    
    Co-authored-by: ᗪєνιη ᗷυнʟ <onedr0p@users.noreply.github.com>
    
    * Update charts/stable/prometheus-nut-exporter/templates/prometheusrules.yaml
    
    Co-authored-by: ᗪєνιη ᗷυнʟ <onedr0p@users.noreply.github.com>
    
    * Update charts/stable/prometheus-nut-exporter/templates/prometheusrules.yaml
    
    Co-authored-by: ᗪєνιη ᗷυнʟ <onedr0p@users.noreply.github.com>
    
    * Update charts/stable/prometheus-nut-exporter/templates/prometheusrules.yaml
    
    Co-authored-by: ᗪєνιη ᗷυнʟ <onedr0p@users.noreply.github.com>
    
    Co-authored-by: ᗪєνιη ᗷυнʟ <onedr0p@users.noreply.github.com>

diff --git a/charts/stable/prometheus-nut-exporter/Chart.yaml b/charts/stable/prometheus-nut-exporter/Chart.yaml
index 42990247..d86452a6 100644
--- a/charts/stable/prometheus-nut-exporter/Chart.yaml
+++ b/charts/stable/prometheus-nut-exporter/Chart.yaml
@@ -2,7 +2,7 @@ apiVersion: v2
 appVersion: 1.1.1
 description: Prometheus NUT Exporter a service monitor to send NUT server metrics to a Prometheus instance.
 name: prometheus-nut-exporter
-version: 4.1.0
+version: 5.0.0
 kubeVersion: ">=1.16.0-0"
 keywords:
 - nut
diff --git a/charts/stable/prometheus-nut-exporter/README.md b/charts/stable/prometheus-nut-exporter/README.md
index 42298b08..026434eb 100644
--- a/charts/stable/prometheus-nut-exporter/README.md
+++ b/charts/stable/prometheus-nut-exporter/README.md
@@ -1,6 +1,6 @@
 # prometheus-nut-exporter
 
-![Version: 4.1.0](https://img.shields.io/badge/Version-4.1.0-informational?style=flat-square) ![AppVersion: 1.1.1](https://img.shields.io/badge/AppVersion-1.1.1-informational?style=flat-square)
+![Version: 5.0.0](https://img.shields.io/badge/Version-5.0.0-informational?style=flat-square) ![AppVersion: 1.1.1](https://img.shields.io/badge/AppVersion-1.1.1-informational?style=flat-square)
 
 Prometheus NUT Exporter a service monitor to send NUT server metrics to a Prometheus instance.
 
@@ -81,7 +81,11 @@ N/A
 | image.pullPolicy | string | `"IfNotPresent"` | image pull policy |
 | image.repository | string | `"hon95/prometheus-nut-exporter"` | image repository |
 | image.tag | string | `"1.1.1"` | image tag |
-| prometheus.serviceMonitor | object | See values.yaml | Enable and configure a Prometheus serviceMonitor for the chart under this key. |
+| metrics.enabled | bool | See values.yaml | Enable and configure a Prometheus serviceMonitor for the chart under this key. |
+| metrics.prometheusRule | object | See values.yaml | Enable and configure Prometheus Rules for the chart under this key. |
+| metrics.prometheusRule.rules | list | See prometheusrules.yaml | Configure additionial rules for the chart under this key. |
+| metrics.serviceMonitor.labels | object | `{}` |  |
+| metrics.serviceMonitor.targets | list | `[]` |  |
 | service | object | See values.yaml | Configures service settings for the chart. |
 
 ## Changelog
@@ -90,6 +94,12 @@ All notable changes to this application Helm chart will be documented in this fi
 
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+### [5.0.0]
+
+#### Changed
+
+- **BREAKING**: Refactored Prometheus metrics section to add rules. Enabling metrics automatically enables the serviceMonitor.
+
 ### [4.0.0]
 
 #### Changed
@@ -116,6 +126,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 
 - use helm-docs
 
+[5.0.0]: #500
 [4.0.0]: #400
 [3.0.0]: #300
 [2.0.0]: #200
diff --git a/charts/stable/prometheus-nut-exporter/README_CHANGELOG.md.gotmpl b/charts/stable/prometheus-nut-exporter/README_CHANGELOG.md.gotmpl
index dfc9f718..0648a405 100644
--- a/charts/stable/prometheus-nut-exporter/README_CHANGELOG.md.gotmpl
+++ b/charts/stable/prometheus-nut-exporter/README_CHANGELOG.md.gotmpl
@@ -9,6 +9,12 @@ All notable changes to this application Helm chart will be documented in this fi
 
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+### [5.0.0]
+
+#### Changed
+
+- **BREAKING**: Refactored Prometheus metrics section to add rules. Enabling metrics automatically enables the serviceMonitor.
+
 ### [4.0.0]
 
 #### Changed
@@ -35,6 +41,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 
 - use helm-docs
 
+[5.0.0]: #500
 [4.0.0]: #400
 [3.0.0]: #300
 [2.0.0]: #200
diff --git a/charts/stable/prometheus-nut-exporter/templates/prometheusrules.yaml b/charts/stable/prometheus-nut-exporter/templates/prometheusrules.yaml
new file mode 100644
index 00000000..b5c561aa
--- /dev/null
+++ b/charts/stable/prometheus-nut-exporter/templates/prometheusrules.yaml
@@ -0,0 +1,55 @@
+{{- if and .Values.metrics.enabled .Values.metrics.prometheusRule.enabled }}
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  name: {{ include "common.names.fullname" . }}
+  labels:
+    {{- include "common.labels" . | nindent 4 }}
+    {{- with .Values.metrics.prometheusRule.labels }}
+    {{- toYaml . | nindent 4 }}
+    {{- end }}
+spec:
+  groups:
+    - name: {{ include "common.names.fullname" . }}
+      rules:
+        - alert: NutExporterAbsent
+          annotations:
+            description: NUT Exporter has disappeared from Prometheus target discovery.
+            summary: NUT Exporter is down.
+          expr: |
+            absent(up{job=~".*{{ include "common.names.fullname" . }}.*"} == 1)
+          for: 5m
+          labels:
+            severity: critical
+        - alert: UpsStatusUnknown
+          annotations:
+            description: UPS {{ "{{ $labels.ups }}" }} is reporting a status of unknown.
+            summary: UPS status unknown.
+          expr: |
+            nut_status{job=~".*{{ include "common.names.fullname" . }}.*"} == 0
+          for: 10s
+          labels:
+            severity: critical
+        - alert: UpsOnBattery
+          annotations:
+            description: UPS {{ "{{ $labels.ups }}" }} has lost power and is running on battery.
+            summary: UPS is running on battery.
+          expr: |
+            nut_status{job=~".*{{ include "common.names.fullname" . }}.*"} == 2
+          for: 10s
+          labels:
+            severity: critical
+        - alert: UpsLowBattery
+          annotations:
+            description: UPS {{ "{{ $labels.ups }}" }} battery is low and the system is getting
+              ready to shutdown.
+            summary: UPS battery is low.
+          expr: |
+            nut_status{job=~".*{{ include "common.names.fullname" . }}.*"} == 3
+          for: 5s
+          labels:
+            severity: critical
+        {{- with .Values.metrics.prometheusRule.rules }}
+        {{- toYaml . | nindent 8 }}
+        {{- end }}
+{{- end }}
diff --git a/charts/stable/prometheus-nut-exporter/templates/servicemonitor.yaml b/charts/stable/prometheus-nut-exporter/templates/servicemonitor.yaml
index bb537398..346c6edd 100644
--- a/charts/stable/prometheus-nut-exporter/templates/servicemonitor.yaml
+++ b/charts/stable/prometheus-nut-exporter/templates/servicemonitor.yaml
@@ -1,11 +1,11 @@
-{{- if .Values.prometheus.serviceMonitor.enabled }}
+{{- if .Values.metrics.enabled }}
 apiVersion: monitoring.coreos.com/v1
 kind: ServiceMonitor
 metadata:
   name: {{ include "common.names.fullname" . }}
   labels:
     {{- include "common.labels" . | nindent 4 }}
-    {{- with .Values.prometheus.serviceMonitor.additionalLabels }}
+    {{- with .Values.metrics.serviceMonitor.labels }}
     {{- toYaml . | nindent 4 }}
     {{- end }}
 spec:
@@ -13,16 +13,16 @@ spec:
     matchLabels:
       {{- include "common.labels.selectorLabels" . | nindent 6 }}
   endpoints:
-  {{- range .Values.prometheus.serviceMonitor.targets }}
-  - port: metrics
-    interval: {{ .interval }}
-    scrapeTimeout: {{ .scrapeTimeout }}
-    path: /metrics
-    params:
-      target:
-      - "{{ .hostname }}:{{ .port }}"
-    relabelings:
-    - sourceLabels: [__param_target]
-      targetLabel: target
-  {{- end }}
+    - port: metrics
+    {{- range .Values.metrics.serviceMonitor.targets }}
+      interval: {{ .interval }}
+      scrapeTimeout: {{ .scrapeTimeout }}
+      path: /metrics
+      params:
+        target:
+          - "{{ .hostname }}:{{ .port }}"
+      relabelings:
+        - sourceLabels: [__param_target]
+          targetLabel: target
+    {{- end }}
 {{- end }}
diff --git a/charts/stable/prometheus-nut-exporter/values.yaml b/charts/stable/prometheus-nut-exporter/values.yaml
index e5763ed1..5413efe7 100644
--- a/charts/stable/prometheus-nut-exporter/values.yaml
+++ b/charts/stable/prometheus-nut-exporter/values.yaml
@@ -33,14 +33,32 @@ service:
         protocol: TCP
         port: 9995
 
-prometheus:
+metrics:
   # -- Enable and configure a Prometheus serviceMonitor for the chart under this key.
   # @default -- See values.yaml
+  enabled: false
   serviceMonitor:
-    enabled: false
     # Specify the list of NUT servers that should be monitored
     targets: []
-    # - hostname: nut-server
-    #   port: 3493
-    #   interval: 30s
-    #   scrapeTimeout: 10s
+      # - hostname: nut-server
+      #   port: 3493
+      #   interval: 30s
+      #   scrapeTimeout: 10s
+    labels: {}
+  # -- Enable and configure Prometheus Rules for the chart under this key.
+  # @default -- See values.yaml
+  prometheusRule:
+    enabled: false
+    labels: {}
+    # -- Configure additionial rules for the chart under this key.
+    # @default -- See prometheusrules.yaml
+    rules: []
+      # - alert: UpsStatusUnknown
+      #   annotations:
+      #     description: UPS {{ "{{ $labels.ups }}" }} is reporting a status of unknown.
+      #     summary: UPS status unknown.
+      #   expr: |
+      #     nut_status == 0
+      #   for: 10s
+      #   labels:
+      #     severity: critical
